### PR TITLE
Use django 5.0

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -57,6 +56,7 @@ django-formtools = "^2.4.1"
 psycopg = "^3.1.9"
 pillow = "^10.1.0"
 django-imagekit = "^5.0.0"
+iso639-lang = "^2.2.3"
```
Comment (norkans7) [inline pyproject.toml:59]: better option than iso-639

File: pyproject.toml [hunk 2]
```diff
@@ -16,9 +16,9 @@ django-mptt = "^0.12.0"
 django-redis = "^4.12.1"
 django-storages = "^1.11.1"
 django-timezone-field = "^6.1.0"
-djangorestframework = "^3.12.2"
+djangorestframework = "^3.15.0"
```
Comment (norkans7) [inline pyproject.toml:19]: DFR that support Django 5.0

File: pyproject.toml (hunk 1 above)
Comment (rowanseymour) [reply pyproject.toml:59]: any more info on why it's better? This isn't related to poetry's problem with packages ending in digits? Apparently they fixed that today https://github.com/python-poetry/poetry/issues/6996#event-12202223132

Comment (norkans7) [reply pyproject.toml:59]: Well, that might have been the issue I was not able to install iso-639 however I found that it has not been maintained to support new python versions 

https://pypi.org/project/iso-639/

https://pypi.org/project/iso639-lang/

and the iso-639-lang simplifies the language matching where we use that

https://github.com/nyaruka/rapidpro/pull/5125/commits/b60330095be839f301ef3a6050732ea8d848ca27#diff-f368e83adf5078b8df8f5220e0713c428d43480583752ed53920070577478577L61-L76

Comment (norkans7) [reply pyproject.toml]: I moved those changes into their own PR https://github.com/nyaruka/rapidpro/pull/5130

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -4,6 +4,10 @@ jobs:
   test:
     name: Test
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10.x", "3.11.x"]
+        django-version: ["4.2.x", "5.0.x"]
```
Comment (rowanseymour) [inline .github/workflows/ci.yml:10]: upgrading Python is separate and we don't need to maintain support for two Django versions... let's just make a PR that switches to 5

**Comment by rowanseymour** [comment]
@norkans7 is there anything holding us back from just switching today to Django 5 ?

**Comment by norkans7** [comment]
@rowanseymour Support for django 5.0 on DRF was released on March 15th and now we have 3.15.1 that fixed a few bugs, I think we can update to use Django 5.0 and DRF 3.15.1

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**